### DISCUSSION TITLE
Interfacing Adafruit 2.8" TFT Touch Shield v2 to Adafruit Metro ESP32-S3

### DISCUSSION THREAD

**Discussion by sai-ydev** [micropython/micropython#16113] [Libraries & Drivers] [answered]
I am trying to get the [Adafruit 2.8" TFT Touch Shield](https://www.adafruit.com/product/1651) working with the [Adafruit Metro ESP32-S3](https://www.adafruit.com/product/5500). I am running a custom MicroPython binary: MicroPython v1.24.0-preview. 

My display is just blank with the backlight on. I confirmed I can get the display working in this combination using the Arduino IDE in C++.

This tells me I am not setting up the SPI interface correctly in MicroPython. I am using the ILI9341 drivers from [micropython-nano-gui](https://github.com/peterhinch/micropython-nano-gui/tree/master).

I tried using both SoftSPI and the hardware SPI interfaces. Using the display schematic, I verified that the pins I use are correct except for the reset pin. Since there is no reset pin for this shield, I used a dummy pin.

I am trying to run the demos in the nano-gui repository's GUI folder. I am sharing the snippets of the color_setup file for the SoftSPI and SPI interfaces. 

SoftSPI interface:

```
from machine import Pin, SoftSPI
import gc

from drivers.ili93xx.ili9341 import ILI9341 as SSD

pdc = Pin(9, Pin.OUT, value=0)
prst = Pin(8, Pin.OUT, value=1)
pcs = Pin(10, Pin.OUT, value=1)
spi = SoftSPI(polarity=0, phase=0, sck=Pin(13), mosi=Pin(11), miso=Pin(12), baudrate=10_000_000)

gc.collect()  # Precaution before instantiating framebuf

ssd = SSD(spi, cs=pcs, dc=pdc, rst=prst)
```
Hardware SPI interface:

```
from machine import Pin, SPI
import gc

from drivers.ili93xx.ili9341 import ILI9341 as SSD

pdc = Pin(9, Pin.OUT, value=0)
prst = Pin(8, Pin.OUT, value=1)
pcs = Pin(10, Pin.OUT, value=1)
spi = SPI(1, sck=Pin(13), mosi=Pin(11), miso=Pin(12), baudrate=10_000_000)

gc.collect()  # Precaution before instantiating framebuf

ssd = SSD(spi, cs=pcs, dc=pdc, rst=prst)
```
I would appreciate any help in configuring the display correctly.

**Comment by peterhinch** [comment, marked as answer]
Does the display use the 6-way connector? If so, the pinout seems to be
* MOSI 42
* MISO 21
* SCK 39

This is just from a quick look at the datasheets. If this doesn't work I'll examine the schematics.

To test the display, please paste the code provided [here](https://github.com/peterhinch/micropython-nano-gui?tab=readme-ov-file#23-verifying-hardware-configuration). If it works as described, you can be confident that everything is OK and that the demos will run.

[EDIT]
Looking at the display datasheet, there are comments [here](https://learn.adafruit.com/adafruit-2-8-tft-touch-shield-v2/stmpe-resistive-pinout#using-capacitive-touch-version-w-slash-older-arduino-1833488) that may suggest that to use the pins you've specified you need to solder a couple of links. Otherwise my pins should work.

**Reply by sai-ydev** [reply]
I am very embarrassed and so sorry. Since it was an Arduino shield, and when I looked at the Metro ESP32-S3's pinouts, I assumed that the SPI pins typically found on a Uno should work here. 

I should have read the display info carefully. It clearly says that the display is connected to the ICSP port. Thank you!

**Reply by peterhinch** [reply]
Glad it's working. If you decide to try [micropython-touch](https://github.com/peterhinch/micropython-touch) it should work: the touch controller is supported.
